### PR TITLE
Coverity Issue: Resource Leak fixed

### PR DESCRIPTION
**Pull request by kansal**
Resource leak fixed. Please review.

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@kansal please add the coverity number to the commit message, this way we can not track it. And while you look at it make sure you put the issue on your name to prevent any double effort.

File: server/src/org/apache/cloudstack/region/RegionsApiUtil.java [hunk 1]
```diff
@@ -186,8 +186,12 @@ protected static UserAccount makeUserAccountAPICall(Region region, String comman
                 XStream xstream = new XStream(new DomDriver());
                 xstream.alias("useraccount", UserAccountVO.class);
                 xstream.aliasField("id", UserAccountVO.class, "uuid");
-                ObjectInputStream in = xstream.createObjectInputStream(is);
-                return (UserAccountVO)in.readObject();
+                try(ObjectInputStream in = xstream.createObjectInputStream(is);) {
+                    return (UserAccountVO)in.readObject();
+                } catch (IOException e) {
+                    s_logger.error(e.getMessage());
+                    return null;
```
Comment (DaanHoogland) [inline server/src/org/apache/cloudstack/region/RegionsApiUtil.java]: this return is not needed. we will fall through to other return. You can opt not to and put another log message in there as well.

**Comment by kansal** [comment]
@DaanHoogland It was the issue that was reported by our internal coverity. The issue reported is:

"noescape: Resource 'in' is not closed or saved in readObject.
CID 17897: Resource leak (RESOURCE_LEAK). leaked_resource: Variable in going out of scope leaks the resource it refers to."

**Comment by DaanHoogland** [comment]
-1 on this. I cannot track any internal tracking system of another company. please if the public coverity doesn't have this make a jira ticket describve the problem and put the ticket id in hte commit message.

**Comment by miguelaferreira** [comment]
Hi @kansal 

If this issue was caught by your internal Coverity scan, wouldn't it also have been cough by the public Coverity scan? Or are these configured differently?

It would be great to use the public IDs all-around, otherwise it will be very confusing. I can imagine many people have their own internal ticketing systems (for example), and if people would just start to use IDs form those internal systems it would be extremely hard to backtrack. Don't you think?

**Comment by kansal** [comment]
@DaanHoogland @miguelaferreira I understand you concerns. I have created the ticket for this in JIRA with ID: CLOUDSTACK-8692. Will keep this in mind while for future commits. Thanks :)

**Comment by miguelaferreira** [comment]
Cool :+1:

**Comment by DaanHoogland** [comment]
Thanks @kansal , please change the commit message to point to it. For instance:
CLOUDSTACK-8692: resource leak found by the internal coverity instance at <company or department or persons kitchen table>

Otherwise I'm all right with it

**Comment by DaanHoogland** [comment]
thanks @kansal LGTM

**Pull request merged**